hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1268,7 +1269,9 @@ private void addParentChildEntityNames(AbstractEntityInsertAction action, BatchI
 				for ( int i = 0; i < propertyValues.length; i++ ) {
 					Object value = propertyValues[i];
 					Type type = propertyTypes[i];
-					addParentChildEntityNameByPropertyAndValue( action, batchIdentifier, type, value );
+					if ( value != null ) {
+						addParentChildEntityNameByPropertyAndValue( action, batchIdentifier, type, value );
+					}
```

Comment:
The above simple code change fixes the regression issues. When the property value is null, we don't need to consider its referential integrity.